buildkite/resource_cluster.go

```diff
@@ -0,0 +1,200 @@
+			},
+		},
+	}
+}
+
+func (c *clusterResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state *clusterResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	r, err := createCluster(
+		c.client.genqlient,
+		c.client.organizationId,
+		state.Name.ValueString(),
+		state.Description.ValueStringPointer(),
+		state.Emoji.ValueStringPointer(),
+		state.Color.ValueStringPointer(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create Cluster",
+			fmt.Sprintf("Unable to create Cluster: %s", err.Error()),
+		)
+		return
+	}
+
+	state.ID = types.StringValue(r.ClusterCreate.Cluster.Id)
+	state.UUID = types.StringValue(r.ClusterCreate.Cluster.Uuid)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (c *clusterResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state clusterResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	r, err := getCluster(c.client.genqlient, c.client.organization, state.UUID.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read Cluster",
+			fmt.Sprintf("Unable to read Cluster: %s", err.Error()),
+		)
+		return
+	}
+
+	updateClusterResourceState(r.Organization.Cluster, &state)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (c *clusterResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state, plan clusterResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id := state.ID.ValueString()
+
+	_, err := updateCluster(
+		c.client.genqlient,
+		c.client.organizationId,
+		id,
+		plan.Name.ValueString(),
+		plan.Description.ValueStringPointer(),
+		plan.Emoji.ValueStringPointer(),
+		plan.Color.ValueStringPointer(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to update Cluster",
+			fmt.Sprintf("Unable to update Cluster: %s", err.Error()),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (c *clusterResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state clusterResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := deleteCluster(c.client.genqlient, c.client.organizationId, state.ID.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to delete Cluster",
+			fmt.Sprintf("Unable to delete Cluster: %s", err.Error()),
+		)
+		return
+	}
+}
+
+func (c *clusterResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	req.ID = ""
```

Comment:
You can't empty out the ID here, its used in the function below to proxy this id through to the UUID of the cluster
```suggestion
```